packages/path-provider/example/lib/main.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+void main() {
+  runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Path Provider',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(title: 'Path Provider'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key, this.title}) : super(key: key);
+  final String title;
+
+  @override
+  _MyHomePageState createState() => new _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  Future<Directory> _tempDirectory;
+  Future<Directory> _appDocumentsDirectory;
+
+  void _requestTempDirectory() {
+    setState(() {
+      _tempDirectory = getTemporaryDirectory();
+    });
+  }
+
+  Widget _buildTempDirectory(
+      BuildContext context, AsyncSnapshot<Directory> snapshot) {
+    if (snapshot.hasError) {
+      return new Padding(
+          padding: const EdgeInsets.all(16.0),
+          child: new Text('Error: ${snapshot.error}'));
```

Comment:
[Nit] Might look nicer with a trailing comma after new Text(...). Similarly below.

Then again, it might not. Up to you.